tfe/data_source_variables.go

```diff
@@ -0,0 +1,178 @@
+package tfe
+
+import (
+	"fmt"
+	"log"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceTFEWorkspaceVariables() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceVariableRead,
+
+		Schema: map[string]*schema.Schema{
+			"variables": &schema.Schema{
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
```

Comment:
Since this schema is defined 3x, it might be nice to just extract it once and reuse it-- I also removed the redundant `&schema.Schema` declaration

```
varSchema := map[string]*schema.Schema{
    "id": {
	    Type:     schema.TypeString,
	    Computed: true,
    },
    "name": {
	    Type:     schema.TypeString,
	    Computed: true,
    },
    "value": {
	    Type:      schema.TypeString,
	    Computed:  true,
	    Sensitive: true,
    },
    "category": {
	    Type:     schema.TypeString,
	    Computed: true,
    },
    "hcl": {
	    Type:     schema.TypeBool,
	    Computed: true,
    },
    "sensitive": {
	    Type:     schema.TypeBool,
	    Computed: true,
    },
}
```